zebra/zapi_msg.c

```diff
@@ -1348,8 +1348,6 @@ static void zread_fec_register(ZAPI_HANDLER_ARGS)
 
 	s = msg;
 	zvrf = vrf_info_lookup(VRF_DEFAULT);
-	if (!zvrf)
```

Comment:
so I don't really see the issue here - why is it wrong to have a check here and there? will removing these checks introduce coverity errors, for example, since coverity can't know the special semantics of the "default" vrf?

Reply:
because default vrf is systematically created.
I think removing this brings clarity?
is it possible to avoid the coverity errors?

Reply:
Before this PR, a few places have already skipped check, like `zebra_vxlan_sg_deref()`. So i think there maybe is no coverity problem.

Reply:
Even these PRs that do not add any capability or functionality take time and energy from the community.

Just for the record, I see this in zebra_vxlan_sg_deref():

```
static void zebra_vxlan_sg_ref(struct in_addr local_vtep_ip,
				struct in_addr mcast_grp)
{
	struct zebra_vrf *zvrf;

[...]

	zvrf = vrf_info_lookup(VRF_DEFAULT);
	if (!zvrf)
		return;

```

There is no benefit to removing these checks. What is the point of even spending the time required to _find_ them?

Reply:
Sorry to take up so much time. I had checked wrong code, please ignore above `zebra_vxlan_sg_deref()`.

Reply:
I'll add my 2c.. I like this change a lot. To answer the question "why?" b/c we are creating an invariant by saying the default VRF is always present. The code then does not have to and in fact should not check for variations in the invariant (e.g., you wouldn't write `int a = 2; assert(a);`).

The reason its so great to eliminate unnecessary checks is that it by definition it reduces the complexity of a program, and less complexity but same functionality is always better -- it's easier to author correctly, easier to understand, easier to debug and easier to change and maintain.